### PR TITLE
Support Python 3.9

### PR DESCRIPTION
**Pull request by stevepiercy**
- Clean up readme

### CODE REVIEW

**Comment by stevepiercy** [comment]
See #115

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,14 +41,71 @@ workflows:
...
+          command: |
+            cd examples/splitfile
+            pipenv run python -m unittest tests.py
+
+      - store_artifacts:
+          path: htmlcov
+
+      - store_artifacts:
+          path: htmltypecov
+
```
Comment (zupo) [inline .circleci/config.yml:100]: This whole block of configuration shouldn't be needed. It seems it is duplicated, but could be taken from `&test-template` ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -36,14 +41,71 @@ workflows:
...
 
+  test_39: &test-template
+    <<: *defaults
+
+    steps:
+      - checkout
+
+      - run:
+          name: Test minimal openapi-core version on Python 3.9
+          command: ./inject_minimal_openapi-core.sh
```
Comment (zupo) [inline .circleci/config.yml:52]: This file probably needs to be modified to add support for another Python version.

**Review by zupo — COMMENTED** [review]
Quick brain dump, I could be wrong on most of my points, just want to get them out there before I have time to dig deeper.

**Comment by zupo** [comment]
@stevepiercy: Can you please do one of the following:

1. Grant me push permissions on your fork so that I can push my changes.
2. Push changes into a branch of `Pylons/pyramid_openapi3` so that I can push my changes. 
3. Commit & push these yourself:

```
diff --git a/.circleci/config.yml b/.circleci/config.yml
index 87e878e..2548cf5 100644
--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -98,67 +98,15 @@ jobs:
       - store_artifacts:
           path: htmltypecov
 
-  test_38: &test-template
+  test_38:
     <<: *defaults
-
-    steps:
-      - checkout
-
-      - run:
-          name: Test minimal openapi-core version on Python 3.8
-          command: ./inject_minimal_openapi-core.sh
-
-      - restore_cache:
-          keys:
-            - v2-dependencies-{{ checksum "Pipfile.lock" }}
-
-      - run:
-          name: install dependencies
-          command: |
-            sudo pip install pipenv
-            pipenv install --dev --deploy
-            touch .installed
-
-      - save_cache:
-          paths:
-            - ./.venv
-          key: v2-dependencies-{{ checksum "Pipfile.lock" }}
-
-      - run:
-          name: run tests
-          command: |
-            make lint all=true
-            make types
-            make unit
-
-      - run:
-          name: run tests for the singlefile example
-          command: |
-            cd examples/singlefile
-            pipenv run python -m unittest app.py
-
-      - run:
-          name: run tests for the todoapp example
-          command: |
-            cd examples/todoapp
-            pipenv run python -m unittest tests.py
-
-      - run:
-          name: run tests for the splitfile example
-          command: |
-            cd examples/splitfile
-            pipenv run python -m unittest tests.py
-
-      - store_artifacts:
-          path: htmlcov
-
-      - store_artifacts:
-          path: htmltypecov
+    docker:
+      - image: circleci/python:3.8
 
   test_37:
     <<: *test-template
     docker:
-      - image: circleci/python:3.7.5
+      - image: circleci/python:3.7
 
   release:
     <<: *defaults
diff --git a/shell.nix b/shell.nix
index ccaa55c..2c32c00 100644
--- a/shell.nix
+++ b/shell.nix
@@ -11,7 +11,7 @@ pkgs.mkShell {
   name = "dev-shell";
   buildInputs = [
     pkgs.pipenv
-    pkgs.python38Full
+    pkgs.python39Full
     pkgs.codespell
   ];
```

**Comment by zupo** [comment]
Same for https://github.com/niteoweb/pyramid-realworld-example-app/pull/145.

File: .circleci/config.yml [hunk 3]
```diff
@@ -36,14 +41,71 @@ workflows:
...
+
   test_38: &test-template
     <<: *defaults
 
     steps:
       - checkout
 
       - run:
-          name: Test minimal openapi-core version on Python 3.7
+          name: Test minimal openapi-core version on Python 3.8
```
[the remark this answers is not in the document]
Comment (stevepiercy) [reply .circleci/config.yml]: @zupo I'm not clear on how to square this change with the above diff. I've sent you an invitation to push to my fork. I reckon that we would need to parameterize the name in the template?

File: .circleci/config.yml (hunk 1 above)
Comment (stevepiercy) [reply .circleci/config.yml:100]: Updated, per @zupo [comment](https://github.com/Pylons/pyramid_openapi3/pull/116#issuecomment-783561291).

File: .circleci/config.yml [hunk 4]
```diff
@@ -36,14 +41,14 @@ workflows:
...
-  test_38: &test-template
+  test_39: &test-template
     <<: *defaults
 
     steps:
       - checkout
 
       - run:
-          name: Test minimal openapi-core version on Python 3.7
+          name: Test minimal openapi-core version on Python 3.9
```
Comment (zupo) [inline .circleci/config.yml:51]:
```suggestion
          name: Test minimal openapi-core version on Python 3.7
```

**Comment by zupo** [comment]
I don't understand why CircleCI's Python3.8 build is failing.

**Comment by stevepiercy** [comment]
Me neither. Let's start with the last successful build on 3.8, and see what has changed.

**Comment by stevepiercy** [comment]
This [commit](https://github.com/Pylons/pyramid_openapi3/pull/116/commits/1f2f0a2ab95f0d21b0969665e5558f643811d119) broke it. The immediately preceding commit was successful.

**Comment by stevepiercy** [comment]
It appears that [CircleCI cannot find pip](https://app.circleci.com/pipelines/github/Pylons/pyramid_openapi3/369/workflows/1ba288c8-7cbd-4f4c-81da-98c90f7b12c4/jobs/650).

```
[InstallError]:       retcode = do_install(
[InstallError]:   File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 2052, in do_install
[InstallError]:       do_init(
[InstallError]:   File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 1304, in do_init
[InstallError]:       do_install_dependencies(
[InstallError]:   File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 899, in do_install_dependencies
[InstallError]:       batch_install(
[InstallError]:   File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 796, in batch_install
[InstallError]:       _cleanup_procs(procs, failed_deps_queue, retry=retry)
[InstallError]:   File "/usr/local/lib/python3.8/site-packages/pipenv/core.py", line 703, in _cleanup_procs
[InstallError]:       raise exceptions.InstallError(c.dep.name, extra=err_lines)
[pipenv.exceptions.InstallError]: Traceback (most recent call last):
[pipenv.exceptions.InstallError]:   File "/home/circleci/repo/.venv/bin/pip", line 5, in <module>
[pipenv.exceptions.InstallError]:     from pip._internal.cli.main import main
[pipenv.exceptions.InstallError]: ModuleNotFoundError: No module named 'pip'
ERROR: Couldn't install package: appdirs
 Package installation failed...
```

WAT?

**Review by stevepiercy — COMMENTED** [review]
Are you kidding me? This is how you have to purge its cache?

**Comment by stevepiercy** [comment]
Indeed, [incrementing the version number is the documented way to clear caches](https://circleci.com/docs/2.0/caching/#clearing-cache).

**Comment by zupo** [comment]
All green after purging CircleCI cache.

**Pull request merged**